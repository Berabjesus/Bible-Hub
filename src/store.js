import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from './reducers';

const middlewares = [thunk];
const store = createStore(rootReducer, applyMiddleware(...middlewares));

export default store;
