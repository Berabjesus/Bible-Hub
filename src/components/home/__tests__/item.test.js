import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Item from "../item";
import { BrowserRouter } from "react-router-dom";
import { Provider } from 'react-redux';
import store from '../../../store'

describe("Item Home component", () => {
  it("matches the snapshot", () => {
    const component = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Item id ={'test-id'} description={'test-description'} image={'test-image'} title={'test-title'} languages={['test-languages-1', 'test-languages-2']}/>
        </Provider>
      </BrowserRouter>
    );
    expect(component).toMatchSnapshot();
  });
  it("Renders the Item component correctly", () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Item id ={'test-id'} description={'test-description'} image={'test-image'} title={'test-title'} languages={['test-languages-1', 'test-languages-2']}/>
        </Provider>
      </BrowserRouter>
    );

    expect(screen.getByAltText('test-title'));

    expect(
      screen.getByText((content, element) => {
        return (
          element.tagName.toLowerCase() === "h4" &&
          content.includes('test-title')
        );
      })
    );

    expect(
      screen.getByText((content, element) => {
        return (
          element.tagName.toLowerCase() === "p" &&
          content.includes('test-description')
        );
      })
    );
  });
});
