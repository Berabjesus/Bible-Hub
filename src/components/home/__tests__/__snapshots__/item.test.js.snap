// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Home component matches the snapshot 1`] = `
<a
  className="d-flex col-12 col-sm-6 col-lg-4 p-0 m-0 wrapper "
  href="/bible/test-id"
  onClick={[Function]}
>
  <div
    className="w-100 h-100 background"
    style={
      Object {
        "background": "url(test-imageUrl) no-repeat center/cover",
      }
    }
  />
  <img
    alt="test-title"
    className="imageUrl mx-auto"
    loading="lazy"
    src="test-imageUrl"
  />
  <div
    className="d-flex flex-column w-100 py-1 font-josefin description "
  >
    <h4>
      test-title
    </h4>
    <p>
      test-description
    </p>
  </div>
</a>
`;
